pkgs/top-level/all-packages.nix

```diff
@@ -10892,20 +10892,24 @@ with pkgs;
 
   grocy = callPackage ../servers/grocy { };
 
-  inherit (callPackages ../servers/nextcloud {})
-    nextcloud27 nextcloud28 nextcloud29;
-
-  nextcloud27Packages = callPackage ../servers/nextcloud/packages {
-    apps = lib.importJSON ../servers/nextcloud/packages/27.json;
+  nextcloud29 = nextcloud;
+  nextcloud28 = nextcloud.overrideAttrs rec {
```

Comment:
So for the sake of having `by-name`, we... make all-packages longer than it was before?

The RFC was supposed to achieve the exact opposite.

:-1: on that.

Reply:
We can keep the old layout and just ignore the failing github action for this PR 😉

Reply:
> We can keep the old layout and just ignore the failing github action for this PR 😉

I think I'm missing something which GH action is failing on this PR?

Reply:
I like the Nextcloud package in `pkgs/by-name/ne/nextcloud` a lot since the `package.nix` file now is very clean and easy to read. Is there any other place where we could define the overrides for the old version 28?

Reply:
> a lot since the package.nix file now is very clean and easy to read

and the way it was before isn't? This is a pretty common pattern in nixpkgs fwiw.

Also, the consequence is the plain ugly solution with `overrideAttrs`: now we're messing around with derivation-level attributes instead of having a clear interface for building Nextcloud derivations.

> where we could define the overrides for the old version 28?

See https://github.com/NixOS/nixpkgs/blob/5ea6f34ee541b37e8a5abe4d7259dfbd8b8170af/pkgs/by-name/README.md?plain=1#L124

A structure outside of by-name is used even then.